PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['chatbot.script']
-    _inherit = ['chatbot.script.step']
-    _inherit = ['discuss.channel']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/crm_livechat/models/chatbot_script.py
+++ b/addons/crm_livechat/models/chatbot_script.py
@@ -5,7 +5,7 @@ from odoo import models, fields
 
 
 class ChatbotScript(models.Model):
-    _inherit = ['chatbot.script']
+    _inherit = 'chatbot.script'
 
     lead_count = fields.Integer(
         string='Generated Lead Count', compute='_compute_lead_count')

--- a/addons/crm_livechat/models/chatbot_script_step.py
+++ b/addons/crm_livechat/models/chatbot_script_step.py
@@ -5,7 +5,7 @@ from odoo import _, models, fields
 
 
 class ChatbotScriptStep(models.Model):
-    _inherit = ['chatbot.script.step']
+    _inherit = 'chatbot.script.step'
 
     step_type = fields.Selection(
         selection_add=[('create_lead', 'Create Lead')], ondelete={'create_lead': 'cascade'})

--- a/addons/crm_livechat/models/discuss_channel.py
+++ b/addons/crm_livechat/models/discuss_channel.py
@@ -7,7 +7,7 @@ from odoo.tools import html2plaintext
 
 
 class DiscussChannel(models.Model):
-    _inherit = ['discuss.channel']
+    _inherit = 'discuss.channel'
 
     def execute_command_lead(self, **kwargs):
         key = kwargs['body']
